applications/cheersland.md

```diff
@@ -214,6 +223,16 @@ Source codes will reside in: https://github.com/cheersland
 * **FTE:**  4
 * **Costs:** $4,000 DAI
 
+**Goal - Implement Multi-asset Staking Platform and the second game.**  
+We will develop the multi-asset staking module and deploy the Staking Contract on Plasm. When users stake tokens as required, the contract will count HPs based on their staking status. All contracts will be released and used after the audit is completed. We will develop the Multi-asset Staking Platform and open several staking pools including DOT and KSM first. Players can earn HPs by staking a certain amount of DOT or KSM and use them in the game.  
+Our game dev team will develop a new classic game for the Polkadot ecosystem, similar to the Crypto Pirates deployed on BSC, which will adopt the same business model and anti-cheating algorithm.  
+
+We will optimize UI and allow players to have a more cheerful gaming experience.  
+**Therefore, under this milestone, our deliverables include:**  
+- Multi-asset Staking Contract 

```

Comment:
Are you using any existing reference implementation for the multi-asset staking contract? As for the previous point, are you going to develop it in Solidity or `ink!`?

Reply:
Actually, this Multi-Asset Platform is an innovative game solution. By staking tokens from the Polkadot ecosystem, players can earn Health Point (HP), the universal ticket for all games based on Cheersland, and it's also the guarantee for the player to get a higher ranking in the game. Therefore, our model is different from other staking platforms and our technical implementation also comes from our original design. We will also use Solidity and it will be rigorously tested and audited by the professional authority agency.

Reply:
Actually, this Multi-Asset Platform is an innovative game solution. By staking tokens from the Polkadot ecosystem, players can earn Health Point (HP), the universal ticket for all games based on Cheersland, and it's also the guarantee for the player to get a higher ranking in the game. Therefore, our model is different from other staking platforms and our technical implementation also comes from our original design. We will also use Solidity and it will be rigorously tested and audited by the professional authority agency.

Reply:
So historically we have been quite reluctant to fund Solidity contracts, for the simple reason that it's close to impossible to verify whether the code written for the grant is original and not a copy of another existing Solidity smart contract. Are you open to using `ink!` instead?

Reply:
No problem sir. Actually, ink! is a Rust-based eDSL, which is very consistent with the technical preference of our team. As a technology-based team, we are intrigued by the technology of efficiency and conciseness. That's why we'd like to deploy our products on Polkadot. We previously chose Solidity due to the time for the development. using  ink! instead, we might increase the time for the development of milestones by one or two weeks, and have resubmitted the commit as below. Glad you suggested using ink! sir.
https://github.com/w3f/Grants-Program/pull/618/commits/3ec78ff92354f1819bb0c3b5a43ae5a3c86ca8de